lib/puppet/util/tcp_validator.rb

```diff
@@ -0,0 +1,36 @@
+require 'socket'
+require 'timeout'
+require 'ipaddr'
+
+module Puppet
+  module Util
+    class TcpValidator
```

Comment:
It would be great to move that to lib/puppetx/puppetlabs, as per https://github.com/puppetlabs/puppet/blob/master/lib/puppet_x.rb